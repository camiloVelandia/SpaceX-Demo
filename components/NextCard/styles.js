import styled from "styled-components";

export const Section = styled.article`
  width: 48%;
  margin-bottom: 40px;
  border: 1px solid white;
  text-align: center;
`;
export const Title = styled.h2`
  background-color: white;
  color: black;
  padding: 30px 0;
  font-size: 36px;
  text-transform: uppercase;
  font-weight: 600;
  margin: 0;
`;

export const Name = styled.h3`
  font-size: 24px;
  margin-top: 30px;
  margin-bottom: 30px;
  text-transform: uppercase;
  padding: 0 20px;
  margin:  0;
  margin-top:40px;
`;
export const Text = styled.p`
  padding: 20px 20px;
  text-align: justify;
  padding-bottom: 30px;
  margin: 0;
`;
export const Info = styled.div`
  padding: 0 20px;
  padding-bottom: 20px;
  display: flex;
  justify-content: space-between;
`;
export const Bullet = styled.div`
  & p {
    margin: 0;
    text-align: left;
  }
`;
