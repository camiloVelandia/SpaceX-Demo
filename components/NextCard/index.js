import React from "react";
import { Section, Title, Figure, Name, Text, Info, Bullet } from "./styles";

const NextCard = (props) => {
  return (
    <Section>
      <Title>Next missions</Title>
      <Figure>
        <img src={props.image || "/images/escudo.png"} />
      </Figure>
      <Name>{props.name}</Name>
      <Text>
        {props.details || 'Details must be ready in the upcomings days, the information is not provides still.'}
      </Text>
      <Info>
        <Bullet>
          <p>date:</p>
          <p>{props.date}</p>
        </Bullet>
        <Bullet>
          <p>flight number:</p>
          <p>{props.number}</p>
        </Bullet>
        
      </Info>
    </Section>
  );
};

export default NextCard;
