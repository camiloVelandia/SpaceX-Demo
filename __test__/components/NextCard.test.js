import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import NextCard from "../../components/NextCard";

describe("<NextCard/>", () => {
  const nextcard = mount(<NextCard />);
  test("render del componente nextcard", () => {
    expect(nextcard.length).toEqual(1);
  });
});

describe("nextcard snaaapshot", () => {
  test("Comprobar ui del nextcard", () => {
    const nextcard = create(<NextCard />);
    expect(nextcard.toJSON()).toMatchSnapshot();
  });
});
