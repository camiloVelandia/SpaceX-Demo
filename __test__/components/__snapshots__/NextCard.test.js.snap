// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nextcard snaaapshot Comprobar ui del nextcard 1`] = `
<article
  className="styles__Section-mhp48x-0 fUrKRR"
>
  <h2
    className="styles__Title-mhp48x-1 jBTTBv"
  >
    Next missions
  </h2>
  <figure
    className="styles__Figure-mhp48x-2 kzVUdW"
  >
    <img
      src="/images/escudo.png"
    />
  </figure>
  <h3
    className="styles__Name-mhp48x-3 eCsUFn"
  />
  <p
    className="styles__Text-mhp48x-4 gJOFuB"
  >
    Details must be ready in the upcomings days, the information is not provides still.
  </p>
  <div
    className="styles__Info-mhp48x-5 bcmowQ"
  >
    <div
      className="styles__Bullet-mhp48x-6 kBZqf"
    >
      <p>
        date:
      </p>
      <p>
        
      </p>
    </div>
    <div
      className="styles__Bullet-mhp48x-6 kBZqf"
    >
      <p>
        flight number:
      </p>
      <p />
    </div>
  </div>
</article>
`;
